File: Solutions/ICE3-Solution/Q1/Commando.java
public class Commando extends Soldier {
    private double vocationAllowance;

    public Commando(String name, double baseAllowance, double vocationAllowance) {
        super(name, baseAllowance);
        this.vocationAllowance = vocationAllowance;
    }

    @Override
    public double getGrossAllowance() {
        return getBaseAllowance() + vocationAllowance;
    }

    @Override
    public String toString() {
        return String.format("Commando{%s, vocationAllowance=%.1f}", super.toString(), vocationAllowance);
    }
}

File: Solutions/ICE3-Solution/Q5/Commando.py
from Soldier import Soldier

class Commando(Soldier):
    def __init__(self, name, base_allowance, vocation_allowance):
        Soldier.__init__(self, name, base_allowance)
        self.vocation_allowance = vocation_allowance

    @property
    def gross_allowance(self):
        return self.base_allowance + self.vocation_allowance

    def __str__(self):
        return "Commando{{{}, vocationAllowance={:.1f}}}".format(Soldier.__str__(self), self.vocation_allowance)
